### ISSUE TITLE
TableHelper macro seems to fail if table contains a method named `insert`

### ISSUE THREAD

**Issue by jalaziz** [outworkers/phantom#766]
I ran into an issue after upgrading from 1.24 to the latest 2.15.

We have two tables that define an `insert` method similar to: `def insert(se: SessionEvent): Future[ResultSet] = insertQuery(se).future()`

The definition doesn't appear to conflict with the Table's start `insert()` method. However, when compiling, we receive the `could not find implicit value for parameter helper: com.outworkers.phantom.macros.TableHelper...` error.

Renaming or removing the method is a workaround, but it's not clear as to why the method breaks the macro.

Here's a simplified schema that fails to compile:
```
import java.util.UUID

import com.outworkers.phantom.dsl._

import scala.concurrent.{Future => ScalaFuture}

case class ClientKey(sessionId: UUID, key: String, other: String)

abstract class ClientKeys extends Table[ClientKeys, ClientKey] {
  object sessionId extends UUIDColumn with PartitionKey
  object key extends StringColumn with ClusteringOrder
  object other extends StringColumn

  def findById(id: UUID): ScalaFuture[Option[ClientKey]] = {
    select.where(_.sessionId eqs id).one()
  }

  def insert(c: ClientKey): ScalaFuture[ResultSet] = {
    insert().future()
  }
}
```

**Comment by alexflav23** [comment]
@jalaziz It is clear to me however it's not a great error report you get out. `TableHelper` autogenerates the store method, which expands a macro in its own compilation unit, which generates code using that very `insert` method.

In short, that means inside the macro you now have code that doesn't compile, because the compiler is not smart enough to figure out the most suitable overloaded definition of `insert` that it needs, so it blows up, but it's not great at reporting why.

The error reporting is a huge problem for us, and it seems there's not much that can be yet done about that. As soon as ScalaMeta offers support for implicit macros, all the macro backend will be moving, in the interim we are living with a compromised solution.

**Comment by jalaziz** [comment]
The odd thing here is that the two `insert` methods have fundamentally different signatures. One takes an argument and the other doesn't. The conflict doesn't make much sense.

Even more odd is that it doesn't apply universally. I have another table with the same pattern and `insert(record)` method that works just fine.

**Comment by alexflav23** [comment]
Hi @jalaziz Apologies for the long overdue follow up on this, it's been fixed in the latest release going out today. The issue was really strange, as the param types were not yet evaluated at the point where the `TableHelper` macro was generating its automated `store` code, it was getting confused between the `insert` method on `CassandraTable` and the `insert` you added. 

I guess there are edge cases in which this doesn't happen, tbh it's more macro API flakiness, as we've witnessed time and time again.

**Comment by alexflav23** [comment]
@jalaziz Closed in #859
